### ISSUE TITLE
Navigation2D (Godot 3.5 beta2) - Error: "sync: Attempted to merge a navigation mesh triangle edge with another already-merged edge", not present in Godot 3.4.3

### ISSUE THREAD

**Issue by Ranoller** [godotengine/godot#59182]
### Godot version

3.5 beta 2

### System information

Windows 7 and 10, GLES2 and GLES 3

### Issue description

Bug submited in https://github.com/godotengine/godot/issues/57254 and supposedly fixed by https://github.com/godotengine/godot/pull/56879 and https://github.com/godotengine/godot/pull/56600 in still present in Navigation2D. Error spams in console, but NavigationPolygon is created and working. Error spams with polygons with hundreds of points, close to each other, when triangulating the big polygon by code. I attach reproduction project.

Error message:

```
E 0:00:00.999   sync: Attempted to merge a navigation mesh triangle edge with another already-merged edge. Either the Navigation's `cell_size` is different from the one used to generate the navigation mesh or `detail/sample_max_error` is too small. This will cause navigation problem.
  <C++ Source>  modules/navigation/nav_map.cpp:605 @ sync()
```

This is not a breaking bug per se, but a inconvenient spam in console message.

Other significant change since new navigation system is that you can´t use Navigation2D in the **_ready()** function when you are building polygons from code, and you have to workaround with the first frame in **_physics_process** to handle "Navigation2D.get_simple_path()". I made the changes in my project because i want to migrate to 3.5, but is a breaking change that should be documented.

### Steps to reproduce

Run attached project in 3.4.3 and 3.5 beta 2 and see output console.

### Minimal reproduction project

[NavigationMeshBug.zip](https://github.com/godotengine/godot/files/8257816/NavigationMeshBug.zip)

**Comment by SkanerSoft** [comment]
3.5 Beta 3 in 3D

```
E 0:00:01.942   sync: Attempted to merge a navigation mesh triangle edge with another already-merged edge. Either the Navigation's `cell_size` is different from the one used to generate the navigation mesh or `detail/sample_max_error` is too small. This will cause navigation problem.
  <Исходный код C++>modules/navigation/nav_map.cpp:605 @ sync()
```

At the same time, I use the basic NavigationMeshInstance without programming, I set everything up in the editor.

**Comment by Scony** [comment]
Well, this won't be fixed too soon I presume as there are some fundamental problems in the implementation/recast library.
For now, you can try increasing `detail/sample_max_error`

**Comment by Ranoller** [comment]
SkanerSoft can try this. I can´t increase **detail/sample_max_error** because this is a property of NavigationMesh, not available in Navigation2D, NavigationPolygonInstance or NavigationPolygon...

**Comment by akien-mga** [comment]
CC @AndreaCatania

**Comment by smix8** [comment]
Need to also backport the more recent Navigation changes from Godot 4.0 that included a lot of edge merging fixes from groud. I can hardly get any navigation merged in Godot 3.5 that works perfectly in Godot 4.0.

**Comment by akien-mga** [comment]
Fixed by #61996.

**Comment by Flavelius** [comment]
Is this fix in 3.5-RC4? If it should be, then i still have this problem in RC4.
